src/VecSim/vec_sim.h

```diff
@@ -199,6 +199,10 @@ void VecSimTieredIndex_GC(VecSimIndex *index);
 bool VecSimIndex_PreferAdHocSearch(VecSimIndex *index, size_t subsetSize, size_t k,
                                    bool initial_check);
 
+void VecSimTieredIndex_AcquireSharedLocks(VecSimIndex *index);
```

Comment:
I suggest changing `VecSimIndex_GetDistanceFrom` to `VecSimIndex_GetDistanceFrom_Unsafe` and mentioning in the function description that the shared locks should be acquired